lib/internal/main/test_runner.js

```diff
@@ -58,6 +58,8 @@ if (shardOption) {
 }
 
 run({ concurrency, inspectPort, watch: getOptionValue('--watch'), setup: setupTestReporters, shard })
-.once('test:fail', () => {
-  process.exitCode = kGenericUserError;
+.on('test:fail', (data) => {
+  if (data.todo === undefined || data.todo === false) {
```

Comment:
Would `data.todo !== true` work here (and in `harness.js`)?

Reply:
it can be a string with the message (also empty string is a valid todo)